src/SipHash.chpl

```diff
@@ -91,23 +115,27 @@ module SipHash {
       const l = D.size;
       if startLocale == endLocale {
         if startLocale == hereLocale {
-          return computeSipHash(c_ptrTo(start), 0..#l, outlen);
+          return computeSipHash(c_ptrTo(start), 0..#l, outlen, numBytes(t));
         } else {
           var a = c_malloc(msg.eltType, l);
-          GET(a, startLocale, getAddr(start), l);
-          var h = computeSipHash(a, 0..#l, outlen);
+          const byteSize = l:size_t * c_sizeof(t);
+          GET(a, startLocale, getAddr(start), byteSize);
+          var h = computeSipHash(a, 0..#l, outlen, numBytes(t));
```

Comment:
Using `numBytes(t)` here but `c_sizeof(t)` 2 lines up. Might be nice to unify?

Reply:
Hmm, how did I know about `c_sizeof(t)` here, I wonder...? Anyway, which of the two idioms would you use?

Reply:
`numBytes` doesn't work for bools, but it does give you Chapel compile time params if you need that. `c_sizeof` should work for any type and that size will be known to the backend compiler so it will fold stuff out with backend optimizations on.

I tend to use `numBytes` if I can just cause it's nice to stick with chapel code, but if I need the size of bools or something I'll use `c_sizeof`